bin/util/autotune.php

```diff
@@ -29,51 +29,60 @@ function bytestostring($amount) {
 		basename(__FILE__)
 	);
-	fputs(STDERR,
-		"Determines the number of PHP-FPM worker processes for given RAM and CPU cores\n\n".
-		"The initial calculation works as follows:\n".
-		'ceil(log_2(RAM_AVAIL / CALC_BASE)) * NUM_CORES * 2 * (CALC_BASE / memory_limit)'."\n\n".
-		'This result is then capped to at most (RAM_AVAIL / memory_limit) processes'."\n\n".
-		"The purpose of this formula is to ensure that:\n".
-		"1) the number of processes does not grow too rapidly as RAM increases;\n".
-		"2) the number of CPU cores is taken into account;\n".
-		"3) adjusting PHP memory_limit has a linear influence on the number of processes\n".
-		'4) the number of processes never exceeds available RAM for given memory_limit'."\n"
-	);
-	fputs(STDERR,
+	fputs($print_help ? STDOUT : STDERR,
+		"Determines the number of PHP-FPM worker processes for given RAM and CPU cores.\n\n".
+		"The result will always be limited to at most (RAM_AVAIL / memory_limit) workers.\n\n".
+		"A second limit, factoring in the number of CPU cores, is calculated as follows:\n".
+		"ceil(log_2(RAM_AVAIL / CALC_BASE)) * NUM_CORES * 2 * (CALC_BASE / memory_limit)\n\n".
+		"The purpose of applying both of these limits is to ensure that:\n".
+		"1) the number of workers does not grow too rapidly as available RAM increases;\n".
+		"2) the number of workers per CPU core remains reasonable;\n".
+		"3) the number of workers never exceeds available RAM for given memory_limit;\n".
+		"4) adjusting PHP memory_limit has a linear influence on the number of workers.\n\n".
 		"Options:\n".
 		"  -b <CALC_BASE>     The PHP memory_limit on which the calculation of the\n".
-		"                     scaling factors should be based. Defaults to '128M'\n".
-		"  -t <DOCUMENT_ROOT> Dir to read '.user.ini' with memory_limit settings from\n".
-		"  -v                 Verbose mode\n\n".
-		"php_value or php_admin_value lines containing memory_limit INI directives from\n".
-		"a PHP-FPM configuration file or dump (php-fpm -tt) can be fed via STDIN.\n\n"
+		"                     scaling factors should be based. Defaults to '128M'.\n".
+		"  -h, --help         Display this help screen and exit.\n".
+		"  -t <DOCUMENT_ROOT> Dir to read '.user.ini' with memory_limit settings from.\n".
+		"  -v                 Be a little more verbose when printing information.\n".
+		"  -vv, --verbose     Be more verbose when printing information.\n\n".
+		"Lines containing php_value or php_admin_value memory_limit INI directives from\n".
+		"a PHP-FPM configuration file or `php-fpm -tt' dump can be fed via STDIN. These\n".
+		"will then be taken into account when determining the effective memory_limit.\n\n"
 	);
-	exit(2);
+	exit($print_help ? 0 : 2);
 }
 
 $ram = stringtobytes($argv[0]); // first arg is the available memory
 fprintf(STDERR, "Available RAM is %s Bytes\n", bytestostring($ram));
 
 $cores = $argv[1];
-fprintf(STDERR, "Number of CPU cores is %d\n", (int)$cores);
+if($verbose) {
```

Comment:
Isn't this always false? Presuming `1` is falsy?

Even if this is correct as written, it might be clearer to write it using the `>= 1` syntax for parity with the other usages?

Reply:
No, 1 is always truthy. So `if($verbose)` is for cases that should trigger on any level of verbosity (equivalent to `if($verbose > 0)`); the more explicit checks are for higher levels (of which there is currently only a second one).

Reply:
I think I read this as a bash conditional where exit code non-zero is falsy.